runtime/sema/access.go

```diff
@@ -339,6 +339,11 @@ func (e EntitlementMapAccess) entitlementImage(entitlement *EntitlementType) (ou
 // defined by the map in `e`, producing a new entitlement set of the image of the
 // arguments.
 func (e EntitlementMapAccess) Image(inputs Access, astRange func() ast.Range) (Access, error) {
+
+	if e.Type == IdentityMappingType {
+		return inputs, nil
+	}
```

Comment:
Unfortunately I am not sure it's quite this simple.  While this works for any mapped member accessed on a reference type, when an entitlement-mapped member is accessed on an owned value, we don't use the `Image` function to compute the set of output entitlements; we just use the entire codomain of the map. However, if the map is the `Identity` map, the codomain is theoretically infinite, so we cannot do that. We need some sensible behavior in this scenario.

One possible solution would be to just produce a reference that is entitled to all the entitlements supported on the output reference type. So, for example given some definition like:

```
struct X {
   access(A) fun foo()
   access(B | C) var s: String
}
struct Y {
   access(Identity) fun getX(): auth(Identity) &X { ... }
}
```
if `getX()` is called on an owned `Y` value, the resulting reference should be authorized to all the entitlements supported by `X`, i.e. `A`, `B`, and `C`.

Reply:
Yeah, you beat me to it 😄. I was writing a comment saying we might need some more work here, but not sure what else is needed. Currently, this has the minimal set of changes needed to unblock the follow-up work on mutability.

Reply:
While adding this, realized it conflicts with the refactoring I did in https://github.com/onflow/cadence/pull/2587.

So I will add it in a separate PR, once both this and 2587 are merged.